lib/dry/configurable/settings.rb

```diff
@@ -23,6 +23,35 @@ def initialize(elements = EMPTY_ARRAY)
         initialize_elements(elements)
       end
 
+      def replace(settings)
+        unless settings.is_a? Dry::Configurable::Settings
+          raise ArgumentError, "settings must be a Dry::Configurable::Settings"
+        end
+
+        settings.each do |setting|
+          self << setting
+        end
+      end
+
+      def merge(settings)
+        unless settings.is_a? Dry::Configurable::Settings
+          raise ArgumentError, "settings must be a Dry::Configurable::Settings"
+        end
+
+        settings.each do |setting|
+          merge_setting(setting)
+        end
+      end
+
+      # @api private
+      def merge_setting(setting)
+        if key?(setting.name)
+          elements[setting.name].merge(setting)
```

Comment:
This branch feels like it might need some more expansive handling. Right now if we had a settings object containing `a`, `b`, and `c` settings, and they're all primitive values, e.g. strings, and we merged in a settings object with just an `a` string, then what I would expect is the `a` from the second settings to _replace_ the first, which won't happen given the current code, AFAICT.

I think we should expand the test examples to encompass cases like these and adjust the implementation to suit.

One way to handle this could be to check whether `elements[setting.name]` is a `Setting::Nested` and then handle either case appropriately.

Reply:
@timriley I'm thinking about this, and got a little tripped up.

Typically merge does replace the fields on the first, like you said.  I'm not exactly sure what I was thinking.
But I'm also having trouble distinguishing the difference between a merge and a deep replace.
Should replace delete fields that don't exist on the settings object that is being replaced?

Also I feel like merge tends to need to be opinionated on merge strategies.
For instance, how would you want to handle the merging of array / hash values? (or should `#merge` replace any value that isn't a Setting?)

Reply:
The newest commit should replace any nested value that isn't a Settings::Nested.
There is no handling currently for merging primitives.

Reply:
I think this is definitely the right call 👍🏼